CONTRIBUTING.md

```diff
@@ -0,0 +1,110 @@
+# Rodio Development Guide
+
+## Quick Start
+
+1. Install [Rust](https://www.rust-lang.org/tools/install)
+2. Clone the repository: `git clone https://github.com/RustAudio/rodio.git`
+3. Navigate to the project: `cd rodio`
+4. Build the project: `cargo build`
+
+## Project Structure
+
+src/:
+- `source/`: Audio source implementations
+- `decoder/`: Audio format decoders
+- `sink/`: Audio playback and mixing
+- `dynamic_mixer/`: Real-time audio mixing
+- `spatial/`: Spatial audio capabilities
+
+## Coding Guidelines
+
+- Follow [Rust API Guidelines](https://rust-lang.github.io/api-guidelines/)
+- Use `rustfmt` for formatting
+- Implement `Source` trait for new audio sources
+- Use `Sink` for playback management
+
+## Common Tasks
+
+### Adding a New Audio Source or Effect
+
+1. Create a new file in `src/source/`
+2. Implement the `Source` trait to define how audio samples are generated or modified
+3. Consider implementing sources like oscillators, noise generators, or effects like amplification, filtering, or distortion
+4. Optimize for performance, especially for real-time processing in effects
+5. Add unit tests/benchmarks where applicable to ensure proper functionality and performance
```

Comment:
maybe switch the order of 4 and 5. 

also maybe work in there something like this:
We recommend adding a test for your feature before adding it, your PR will need it and it makes development significantly easier. Do not worry about performance or code quality when getting your feature to work. When your feature works, congrats! :tada:. Open a draft PR now as we love helping out during refactoring and optimizing. Then refactor add a benchmark and lets get the implementation fast.

Reply:
Sure, I paraphrased this and worked it in

should we also mention that some people might not know how to write tests, so maybe I should phrase it as a preferred but not enforced thing, as making them add them could add quite the barrier to entry when they could implement it and then work with us to get the tests in (if applicable)

Reply:
> Sure, I paraphrased this and worked it in

thanks!

> maybe I should phrase it as a preferred but not enforced thing, as making them add them could add quite the barrier to entry when they could implement it and then work with us to get the tests in (if applicable)

yeah your right there. Ill see if I can work in some links to good resources though

Reply:
you suggested text is perfect